Src/BootCamp.Chapter/FileCleaner.cs

```diff
@@ -13,8 +14,48 @@ public static class FileCleaner
         /// <param name="dirtyFile">Dirty file with "_" placed in random places.</param>
         /// <param name="cleanedFile">Cleaned up file without any "_".</param>
         public static void Clean(string dirtyFile, string cleanedFile)
+        {            
+            var corruptedFileContent = File.ReadAllText(dirtyFile);
+            if (String.IsNullOrEmpty(corruptedFileContent))
+            {
+                File.WriteAllText(cleanedFile, corruptedFileContent);
+                return;
+            }
+
+            var cleanFileContent = corruptedFileContent.Replace("_", "");
+
+            try
+            {
+                ValidateName(cleanFileContent);
+                ValidateBalance(cleanFileContent);
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidBalancesException("Invalid name or Balance format", ex);
+            }
+
+            File.WriteAllText(cleanedFile, cleanFileContent);
+        }
+
+        public static void ValidateBalance(string fileContent)
+        {
+            var peopleAndBalances = fileContent.Split(Environment.NewLine);
+            for (int i = 0; i < peopleAndBalances.Length; i++)
+            {
+                BalanceStats.ConvertToBalanceArray(peopleAndBalances[i]);
+            }
+        }
+
+        public static void ValidateName(string fileContent)
         {
-            File.WriteAllText(cleanedFile, "a");
+            var name = BalanceStats.ConvertToArray(fileContent)[0];
+            foreach (char letter in name)
+            {
+                if (!Char.IsLetter(letter) && !letter.Equals(' '))
+                {
+                    throw new Exception();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['System.Exception' should not be thrown by user code.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=5012948)